src/mol-plugin-ui/sequence/sequence.tsx

```diff
@@ -199,9 +217,17 @@ export class Sequence<P extends SequenceProps> extends PluginUIComponent<P> {
 
     protected getBackgroundColor(seqIdx: number) {
         const seqWrapper = this.props.sequenceWrapper;
-        if (seqWrapper.isHighlighted(seqIdx)) return this.markerColors.highlighted;
-        if (seqWrapper.isSelected(seqIdx)) return this.markerColors.selected;
-        if (seqWrapper.isFocused(seqIdx)) return this.markerColors.focused;
+        if (seqWrapper.isHighlighted(seqIdx) && this.markerColors.highlighted) return this.markerColors.highlighted;
+        if (seqWrapper.isSelected(seqIdx) && this.markerColors.selected) return this.markerColors.selected;
+        if (seqWrapper.isFocused(seqIdx) && this.markerColors.focused) return this.markerColors.focused;
+        if (this.customColorFunction) {
+            const loci = seqWrapper.getLoci(seqIdx);
+            const location = StructureElement.Loci.getFirstLocation(loci);
```

Comment:
Create a private member `private themeLocation = StructureElement.Location.create()` and do

```suggestion
            const location = StructureElement.Loci.getFirstLocation(loci, this.themeLocation );
```

This will save a lot of allocations.

Reply:
Thanks, will do.